src/sources/http.rs

```diff
@@ -119,6 +148,16 @@ impl SourceConfig for SimpleHttpConfig {
     }
 }
 
+fn add_path(mut events: Vec<Event>, key: &str, path: &str) -> Vec<Event> {
+    for event in events.iter_mut() {
+        event
+            .as_mut_log()
+            .insert(key, Value::from(path.to_string()));
```

Comment:
FWIW this should be equivalent to:
```suggestion
            .insert(key, path.to_string().into());
```
(Not really a suggestion to change it, just a note)